activiti-engine/src/main/java/org/activiti/engine/impl/persistence/entity/TaskEntityManagerImpl.java

```diff
@@ -122,7 +122,7 @@ private void changeTaskAssignee(TaskEntity taskEntity, String assignee, boolean
       if (taskEntity.getId() != null) {
         getHistoryManager().recordTaskAssigneeChange(taskEntity.getId(), taskEntity.getAssignee());
         addAssigneeIdentityLinks(taskEntity);
-        update(taskEntity);
+        update(taskEntity, fireEvents);
```

Comment:
The update method was invoked without fireEvents flag, which was causing the task update entity event fired before task created in the engine.